File: my-solutions/0009-palindrome-number/solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int num ;
        int rev = 0;
        num = x;
        if(x<0) return false;
        while (x != 0) {
            int rem = x % 10;
            if (rev > INT_MAX / 10 || rev == INT_MAX && rem == 7)
                return 0;
            if (rev < INT_MIN / 10 || rev == INT_MIN && rem == -8)
                return 0;
            rev = rev * 10 + (x % 10);
            x = x / 10;
        }

        if(num != rev)
            return false;
        else 
            return true;
    }
};

File: my-solutions/0009-palindrome-number/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:

        num = str(x)

        p1 = 0
        p2 = len(num)-1

        while(p2 > p1):
            if(num[p1] != num[p2]):
                return False

            p1 += 1
            p2 -= 1

        return True
        

        
